_episodes/04-use-license.md

```diff
@@ -5,7 +5,185 @@ exercises: 0
+
+
+## Tell why is important that a product/code has a licence
+
+* Why you need a licence
+* Explain the differences between concepts such as [open](https://opensource.org/docs/osd) and proprietary
+
+## Tell what is the importance of third-party dependencies on your product/code
+
+Third-party dependencies are any library that you are using in your project that was wrote by someone else. For example, if your data analysis in R uses ggplot2 for the visualisation, ggplot2 would be a third-party dependence. Third-party dependencies have their own licenses that can be different from the license that you apply to your project but they should be compatible. The freedom provided by your license must not violate the restriction described in the license of the third-party depencency that you are using. For example:
+
+The [GNU Affero General Public License](https://www.gnu.org/licenses/agpl-3.0.en.html) (AGPL) license says
+
+> if you modify the Program, your modified version must prominently offer all users interacting with it remotely through a computer network (if your version supports such interaction) an opportunity to receive the Corresponding Source of your version by providing access to the Corresponding Source from a network server at no charge, through some standard or customary means of facilitating copying of software.
+
+Therefore, if one of third-party libraries that you are using on your web app is licensed under this license, you should be aware that you will need to provide a copy of this AGPL third-party library that you are using.
+
+Most of the open source licenses can be divided into two big groups: copyleft (or share-alike) and permissive. Copyleft licenses includes a clause that guarantees continued open access to a software and its source code in derivative works. The [GNU General Public License](https://www.gnu.org/licenses/gpl-3.0.en.html) is the most famous copyleft license and it says
+
+> To protect your rights, we need to prevent others from denying you these rights or asking you to surrender the rights. Therefore, you have certain responsibilities if you distribute copies of the software, or if you modify it: responsibilities to respect the freedom of others.
+>
+> For example, if you distribute copies of such a program, whether gratis or for a fee, you must pass on to the recipients the same freedoms that you received. You must make sure that they, too, receive or can get the source code. And you must show them these terms so they know their rights.
+
+Permissive licenses, as opposed to copyleft licenses, does not enforce that derivative works are licensed under the same terms or compatible terms of the original work. Famous examples of permissive licenses are the [MIT](https://opensource.org/licenses/MIT) and [BSD](https://opensource.org/licenses/BSD).
+
+![Diagram](http://journals.plos.org/ploscompbiol/article/figure/image?size=large&download=&id=10.1371/journal.pcbi.1002598.g002)
```

Comment:
This is an awesome figure! Can you add an image credit?

Reply:
I would avoid the term "open access" here, as it's something different, and most versions of open access (notably green and gold open access) are not at all equivalent to open source. "continued access to the software and its source code" would be better.